lib/cn/kvset.c

```diff
@@ -2210,8 +2210,13 @@ kvset_iter_kblock_read(struct work_struct *rock)
     if (ev(err))
         goto done;
 
+    /* [HSE_REVISIT] kr->pc belongs to cn, this thread may be running after cn_close() and
+     * accessing kr->pc would cause a segfault. Uncomment after fixing this.
+     */
+#if 0
     perfc_inc(kr->pc, PERFC_RA_CNCOMP_RREQS);
     perfc_add(kr->pc, PERFC_RA_CNCOMP_RBYTES, iov.iov_len);
+#endif
```

Comment:
What thread is getting caught here?  A compaction thread?  They should have a cn ref, so must be some other thread...

Reply:
the cn_io thread executes this, but it's kicked off by a compaction thread.

Reply:
I think the problem is that cn_io_wq is shared with all trees, so they don't have a ref on cn.  We don't want to flush it in cn_close() because that could stall cn_close() unnecessarily for quite a while if there are other active trees.  I think the only answer here is that async jobs put on the io queue must hold a cn-ref and drop it when they're done.  In this case I think you'd acquire the ref when you set up the iterator, and drop it before you destroy the iterator, then you wouldn't be thrashing the cn-ref on every async job.

But we can do that in a separate ticket.